addOns/pscan/src/main/java/org/zaproxy/addon/pscan/internal/PassiveScannerOptions.java

```diff
@@ -0,0 +1,327 @@
+                            scanner.getResponseBodyRegex());
+            getConfig()
+                    .setProperty(
+                            elementBaseKey + AUTO_TAG_SCANNER_ENABLED_KEY, scanner.isEnabled());
+        }
+    }
+
+    public List<RegexAutoTagScanner> getAutoTagScanners() {
+        return autoTagScanners;
+    }
+
+    @ZapApiIgnore
+    public boolean isConfirmRemoveAutoTagScanner() {
+        return this.confirmRemoveAutoTagScanner;
+    }
+
+    @ZapApiIgnore
+    public void setConfirmRemoveAutoTagScanner(boolean confirmRemove) {
+        this.confirmRemoveAutoTagScanner = confirmRemove;
+        getConfig().setProperty(CONFIRM_REMOVE_AUTO_TAG_SCANNER_KEY, confirmRemoveAutoTagScanner);
+    }
+
+    /**
+     * Sets whether or not the passive scan should be performed only on messages that are in scope.
+     *
+     * @param scanOnlyInScope {@code true} if the scan should be performed only on messages that are
+     *     in scope, {@code false} otherwise.
+     * @see #isScanOnlyInScope()
+     * @see org.parosproxy.paros.model.Session#isInScope(String) Session.isInScope(String)
+     */
+    public void setScanOnlyInScope(boolean scanOnlyInScope) {
+        this.scanOnlyInScope = scanOnlyInScope;
+        getConfig().setProperty(SCAN_ONLY_IN_SCOPE_KEY, scanOnlyInScope);
+    }
+
+    /**
+     * Tells whether or not the passive scan should be performed only on messages that are in scope.
+     *
+     * @return {@code true} if the scan should be performed only on messages that are in scope,
+     *     {@code false} otherwise.
+     * @see #setScanOnlyInScope(boolean)
+     */
+    public boolean isScanOnlyInScope() {
+        return scanOnlyInScope;
+    }
+
+    /**
+     * Sets whether or not the passive scan should be performed on traffic from the fuzzer.
+     *
+     * @param scanFuzzerMessages {@code true} if the scan should be performed on traffic generated
+     *     by the fuzzer, {@code false} otherwise.
+     * @see #isScanFuzzerMessages()
+     */
+    public void setScanFuzzerMessages(boolean scanFuzzerMessages) {
+        this.scanFuzzerMessages = scanFuzzerMessages;
+        getConfig().setProperty(SCAN_FUZZER_MESSAGES_KEY, scanFuzzerMessages);
+        setFuzzerOptin(scanFuzzerMessages);
+    }
+
+    /**
+     * Adds or removes the {@code HistoryReference} types that should be included when passive
+     * scanning traffic from the fuzzer.
+     *
+     * @param shouldOptin {@code true} if the types should be enabled (added) {@code false} if the
+     *     types should be disabled (removed)
+     * @see #isScanFuzzerMessages()
+     * @see #setScanFuzzerMessages(boolean)
+     */
+    private void setFuzzerOptin(boolean shouldOptin) {
+        if (shouldOptin) {
+            PassiveScanTaskHelper.addApplicableHistoryType(HistoryReference.TYPE_FUZZER);
+            PassiveScanTaskHelper.addApplicableHistoryType(HistoryReference.TYPE_FUZZER_TEMPORARY);
```

Comment:
I wonder why I didn't implement a version of this method that accepted an list or varg